src/main/connection_candidate.rs

```diff
@@ -103,7 +104,8 @@ impl<UID: Uid> ConnectionCandidate<UID> {
     fn done(&mut self, core: &mut Core, poll: &Poll) {
         let _ = core.remove_state(self.token);
         let token = self.token;
-        let socket = mem::replace(&mut self.socket, Socket::default());
+        let socket = mem::replace(&mut self.socket, Default::default());
+        let _ = poll.reregister(&socket, token, Ready::readable(), PollOpt::edge());
```

Comment:
hmm actually i want to get rid of this idiom of having to re-register to express exactly what we need etc. It's a laborious way to overly fine-tune and is prone to errors. The uniform idiom should be to just read in a loop. If you collect many msgs and only one is meant to be handled by the current state, then pass the remainder over to the next state. Similarly never re-register to write (or avoid it unless impossible to do without it). Just write instead. Start with registering once for read and write with edge notification and work your way from there.

This will however need a well thought out change so it's OK to do it in future PR. I'll put it as an issue.

Reply:
[issue created here](https://github.com/maidsafe/crust/issues/1087) for future.